### PR TITLE
Add support for Hive thermostat SLR1b and remote WPT1

### PR DESCRIPTION
**Pull request by dmpolukhin**
This pull request is not fully ready yet. getWeeklySchedule support is functional (i.e. I can get and set weekly schedule) but most probably it will require more work. I published it here to discuss implementation details. Another part of the required changes in https://github.com/Koenkk/zigbee-herdsman/pull/128

As far as I understand getWeeklySchedule  is very different from other requests because thermostat responses with multiple messages and I don't know how the data should be presented in Z2M. I choose dictionary of the day to payload. Also perhaps it make sense to decode thermoseqmode instead of using raw message format. Another problem that I see timeout in log:
```
zigbee2mqtt:error 2020-01-18 14:36:15: Publish 'get' 'weekly_schedule' to '0x001e5e0902478585' failed: 'Error: Timeout - 25726 - 5 - 4 - 513 - 11 after 15000ms'
zigbee2mqtt:debug 2020-01-18 14:36:15: Error: Timeout - 25726 - 5 - 4 - 513 - 11 after 15000ms                                                                                                        
  at Timeout.object.timer.setTimeout [as _onTimeout] (/home/pi/zigbee-herdsman/dist/utils/waitress.js:46:24)                                                                                        
  at ontimeout (timers.js:436:11)                                                                                                                                                                   
  at tryOnTimeout (timers.js:300:5)                                                                                                                                                                 
  at listOnTimeout (timers.js:263:5)                                                                                                                                                                
  at Timer.processTimers (timers.js:223:10)    
```
The timeout seems to cause no harm but it something that  needs to be fixed.

### CODE REVIEW

File: devices.js [hunk 1]
```diff
@@ -2386,6 +2404,40 @@ const devices = [
...
+        description: 'Heating thermostat',
+        supports: 'thermostat, occupied heating, weekly schedule',
+        fromZigbee: [fz.thermostat_att_report, fz.thermostat_weekly_schedule_rsp],
+        toZigbee: [
+            tz.thermostat_local_temperature, tz.thermostat_system_mode, tz.thermostat_running_state,
+            tz.thermostat_occupied_heating_setpoint, tz.thermostat_control_sequence_of_operation,
+            tz.thermostat_weekly_schedule, tz.thermostat_clear_weekly_schedule,
+            tz.thermostat_temperature_setpoint_hold, tz.thermostat_temperature_setpoint_hold_duration,
+        ],
+        meta: {configureKey: 4},
```
Comment (Koenkk) [inline devices.js]: To avoid the 15000 ms timeout change this to `meta: {configureKey: 4, options: {disableDefaultResponse: true}},`

Comment (dmpolukhin) [reply devices.js]: Thank you it helps!

File: converters/toZigbee.js [hunk 2]
```diff
@@ -660,15 +660,27 @@ const converters = {
         key: ['weekly_schedule'],
         convertSet: async (entity, key, value, meta) => {
             const payload = {
-                temperature_setpoint_hold: value.temperature_setpoint_hold,
```
Comment (dmpolukhin) [inline converters/toZigbee.js]: As far as I can see code that was here before cannot work because payload format doesn't match setWeeklySchedule command (from the name it looks like it was hvacThermostat attributes instead).

File: converters/toZigbee.js [hunk 3]
```diff
@@ -660,15 +660,27 @@ const converters = {
...
+                if (typeof elem['coolSetpoint'] == 'number') {
+                    elem['coolSetpoint'] = Math.round(elem['coolSetpoint'] * 100);
+                }
+            }
             await entity.command('hvacThermostat', 'setWeeklySchedule', payload, getOptions(meta));
         },
         convertGet: async (entity, key, meta) => {
-            await entity.command('hvacThermostat', 'getWeeklySchedule', {}, getOptions(meta));
+            const payload = {
+                daystoreturn: 0xff, // Sun-Sat and vacation
```
Comment (dmpolukhin) [inline converters/toZigbee.js]: In the ideal world I would get these params from mqtt request but value is not passed here. So I hard-coded "get everything".

File: converters/toZigbee.js [hunk 4]
```diff
@@ -683,33 +695,6 @@ const converters = {
             await entity.command('hvacThermostat', 'getRelayStatusLog', {}, getOptions(meta));
         },
     },
-    thermostat_weekly_schedule_rsp: {
```
Comment (dmpolukhin) [inline converters/toZigbee.js]: Responses weekly_schedule_rsp and relay_status_log_rsp in toZigbee list don't work so I removed them. Unfortunately my thermostat doesn't support relay_status_log so I was not able to test it but weekly_schedule fully tested.

**Comment by Koenkk** [comment]
Can this MR be merged?

**Comment by dmpolukhin** [comment]
> Can this MR be merged?

@Koenkk I think yes, it can be merged. My comments just explained why I changed code outside of thermostat that I'm going to support.

File: devices.js [hunk 5]
```diff
@@ -2405,6 +2422,49 @@ const devices = [
...
+    {
+        zigbeeModel: ['WPT1'],
+        model: 'WPT1',
+        vendor: 'Hive',
+        description: 'Heating thermostat remote control',
+        supports: 'none, communicate via thermostat',
+        fromZigbee: [],
+        toZigbee: [],
+        meta: {configureKey: 1},
+        configure: async (device, coordinatorEndpoint) => {
```
Comment (Koenkk) [inline devices.js]: Last question; Why is this here? I don't understand:

> are required to have device "configured".

Comment (dmpolukhin) [reply devices.js]: It is my mistake. It seems that on clean setup it is not required. I removed it from my patch.

**Comment by Koenkk** [comment]
Looks good, thanks again :100:
